### PR TITLE
Add Cisco Err-Disable interface event trap handler

### PR DESCRIPTION
**Pull request by bonzo81**
Please give a short description what your pull request is for

SNMP Trap hander and test for Cisco Err-Disable Interface Event (Rev1) trap. Requires CISCO-ERR-DISABLE-MIB.

thanks to @Jellyfrog for help with test

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by bonzo81** [comment]
Although the handler and test work, I wasn't sure if my method to extract the ifIndex from the trap using `toString` then `preg_match` was the best way to do that. I looked at using `substr` but couldn't see how to use that when the required string could be single or multiple characters. A review of that would be appreciated.

File: LibreNMS/Snmptrap/Handlers/CiscoErrDisableInterfaceEvent.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     *
+     * @param  Device  $device
+     * @param  Trap  $trap
+     * @return void
+     */
+    public function handle(Device $device, Trap $trap)
+    {
+        $oidText = $trap->toString(true);
+
+        preg_match('/.cErrDisableIfStatusCause\.(.*?)\./', $oidText, $ifIndex);
```
Comment (Jellyfrog) [inline LibreNMS/Snmptrap/Handlers/CiscoErrDisableInterfaceEvent.php:47]: You should replace `toString()` with `$trap->findOid('CISCO-ERR-DISABLE-MIB::cErrDisableIfStatusCause')`

How you want to get the ifIndex is up to you, one way would  be;
```
>>> explode(".", $trap->findOid('CISCO-ERR-DISABLE-MIB::cErrDisableIfStatusCause'))
=> [
     "CISCO-ERR-DISABLE-MIB::cErrDisableIfStatusCause",
     "858734311",
     "0",
   ]
```

Comment (Jellyfrog) [inline LibreNMS/Snmptrap/Handlers/CiscoErrDisableInterfaceEvent.php:47]: You should replace `toString()` with `$trap->findOid('CISCO-ERR-DISABLE-MIB::cErrDisableIfStatusCause')`

How you want to get the ifIndex is up to you, one way would  be;
```
>>> explode(".", $trap->findOid('CISCO-ERR-DISABLE-MIB::cErrDisableIfStatusCause'))
=> [
     "CISCO-ERR-DISABLE-MIB::cErrDisableIfStatusCause",
     "858734311",
     "0",
   ]
```

File: LibreNMS/Snmptrap/Handlers/CiscoErrDisableInterfaceEvent.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+     * Handle snmptrap.
+     * Data is pre-parsed and delivered as a Trap.
+     *
+     * @param  Device  $device
+     * @param  Trap  $trap
+     * @return void
+     */
+    public function handle(Device $device, Trap $trap)
+    {
+        $trap->findOid('CISCO-ERR-DISABLE-MIB::cErrDisableIfStatusCause');
```
Comment (Jellyfrog) [inline LibreNMS/Snmptrap/Handlers/CiscoErrDisableInterfaceEvent.php:45]: this line does nothing

**Review by laf — APPROVED** [review]
LGTM

**Comment by laf** [comment]
Great work @bonzo81 👍

**Pull request merged**